c10/util/LeftRight.h

```diff
@@ -28,6 +28,24 @@ struct IncrementRAII final {
 
 // LeftRight wait-free readers synchronization primitive
 // https://hal.archives-ouvertes.fr/hal-01207881/document
+//
+// LeftRight is quite easy to use (it can make an arbitrary
+// data structure permit wait-free reads), but it has some
+// particular performance characteristics you should be aware
+// of if you're deciding to use it:
+//
+//  - Reads still incur an atomic write (this is how LeftRight
```

Comment:
"write to an atomic counter", maybe, to make clear that it's not writing to the data structure itself